Actions/ProcessALCodeAnalysisLogs/ProcessALCodeAnalysisLogs.ps1

```diff
@@ -13,6 +13,73 @@ if (Test-Path $sarifPath) {
     OutputError -message "Base SARIF file not found at $sarifPath"
 }
 
+<#
+    .SYNOPSIS
+    Finds a file in the workspace based on its absolute path.
+    .DESCRIPTION
+    Given an absolute path, this function searches the workspace for a file with the same name and returns its relative path.
+    If multiple files with the same name are found, it returns the one with the longest matching suffix to the absolute path.
+    If no file is found, it returns $null.
+    .PARAMETER AbsolutePath
+    The absolute path of the file to find.
+    .PARAMETER WorkspacePath
+    The workspace path to search in. Defaults to the GITHUB_WORKSPACE environment variable.
+#>
+function Get-FileFromAbsolutePath {
```

Comment:
I'd really appreciate a couple of tests for this function.

Reply:
That's also exactly why this PR is still in draft 👍 More tests coming!